cmd/vicadmin/vicadm.go

```diff
@@ -483,6 +483,9 @@ func vSphereSessionGet(sessconfig *session.Config) (*session.Session, error) {
 		log.Errorf("Got %s while creating user session", err)
 		return nil, err
 	}
+	if usersession == nil {
+		return nil, fmt.Errorf("UserSession from vSphere came back nil but no error was reported")
```

Comment:
Not sure why you chose this error message.. as we discussed and as is documented, UserSession returns nil, nil if the session is no longer valid.  At which point you need to `Login()` to get a new session.

https://github.com/vmware/govmomi/blob/master/session/manager.go#L133-L134